subprojects/core/src/main/java/org/gradle/api/internal/classpath/DefaultModuleRegistry.java

```diff
@@ -161,6 +162,17 @@ private Module loadOptionalModule(String moduleName) {
 
     private Module module(String moduleName, Properties properties, Set<File> implementationClasspath) {
         String[] runtimeJarNames = split(properties.getProperty("runtime"));
+        if (JavaVersion.current().compareTo(JavaVersion.VERSION_1_9) < 0) {
+            List<String> filteredRuntimeJarNames = new ArrayList<String>(runtimeJarNames.length);
+            for (int i = 0; i < runtimeJarNames.length; i++) {
+                if (!runtimeJarNames[i].startsWith("jaxb-impl-")) {
```

Comment:
I don't like that this is hard-coded here. Instead I thought about writing some kind of metadata attribute to `${moduleName}-classpath.properties`. However, since it's only used for this dependency that might be over-engineered.

Reply:
I think for now this is fine, but agree we should model this better if we encounter more such problems.